cookbooks/aws-parallelcluster-config/recipes/imds.rb

```diff
@@ -15,7 +15,7 @@
 # OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions and
 # limitations under the License.
 
-return if node['cluster']['scheduler'] == 'awsbatch'
+return if virtualized? || node['cluster']['scheduler'] == 'awsbatch'
```

Comment:
With this new check we're skipping the execution of this code in the systemtests too, instead before they were executed. This let me think we can mock something and execute it in docker too.

Reply:
I found the point. We were skipping it in the `init.rb`

```
# IMDS
include_recipe 'aws-parallelcluster-config::imds' unless virtualized?
```

We can remove the `unless` check from init.rb and keep this one you added.